### PR TITLE
Adds hook_mode_create_access.

### PR DESCRIPTION
**Pull request by agentrickard**
See #58

### CODE REVIEW

File: workbench_access.module [hunk 1]
```diff
@@ -168,6 +168,35 @@ function workbench_access_node_access(NodeInterface $node, $op, AccountInterface
...
+  $manager = \Drupal::service('plugin.manager.workbench_access.scheme');
+  $scheme = $manager->getActiveScheme();
+  if (!$scheme) {
+    return $return;
+  }
+
+  // Check that access control applies to this content type.
+  $type = \Drupal::entityTypeManager()->getStorage('node_type')->load($entity_bundle);
+  $active = $type->getThirdPartySetting('workbench_access', 'workbench_access_status', 0);
+  if (!$active) {
```
Comment (larowlan) [inline workbench_access.module]: nit: is there a need for the intermediate $active variable?

File: workbench_access.module [hunk 2]
```diff
@@ -168,6 +168,35 @@ function workbench_access_node_access(NodeInterface $node, $op, AccountInterface
...
+ * Implements hook_node_create_access().
+ *
+ * @link https://www.drupal.org/node/2348203
+ */
+function workbench_access_node_create_access(AccountInterface $account, $context, $entity_bundle) {
+  $return = AccessResult::neutral();
+  // Check that our access control is configured.
+  $manager = \Drupal::service('plugin.manager.workbench_access.scheme');
+  $scheme = $manager->getActiveScheme();
+  if (!$scheme) {
```
Comment (larowlan) [inline workbench_access.module]: nit: is there a need for the intermediate $scheme variable?

File: workbench_access.module [hunk 3]
```diff
@@ -168,6 +168,35 @@ function workbench_access_node_access(NodeInterface $node, $op, AccountInterface
...
+
+  // Check that access control applies to this content type.
+  $type = \Drupal::entityTypeManager()->getStorage('node_type')->load($entity_bundle);
+  $active = $type->getThirdPartySetting('workbench_access', 'workbench_access_status', 0);
+  if (!$active) {
+    return $return;
+  }
+
+  // Check that the user is able to assign content to a section.
+  $user_sections = $manager->getUserSections($account->id());
```
Comment (larowlan) [inline workbench_access.module]: same here

**Review by larowlan — COMMENTED** [review]
I'll have a go at some tests to go with it today.

As, the need for this issue indicates the feature doesn't work as is, we should add a test to ensure it continues to work.

File: workbench_access.module (hunk 2 above)
Comment (agentrickard) [reply workbench_access.module]: Just habit, really.

Comment (agentrickard) [reply workbench_access.module]: Actually, I'm always uncertain on the negation syntax in that IF case.

It's not `if ($scheme 1= $manager->getActiveScheme())` and I don't trust `if (!$scheme = $manager->getActiveScheme())` to read clearly.

**Comment by larowlan** [comment]
all green :)

**Pull request merged**